tests/foreman/api/test_contentviewversion.py

```diff
@@ -33,960 +33,1024 @@
+    # sync repository
+    repo.sync()
+    # Create content view
+    content_view = entities.ContentView(organization=org).create()
+    # Associate repository to new content view
+    content_view.repository = [repo]
+    content_view = content_view.update(['repository'])
+    # Publish content view
+    content_view.publish()
+    content_view = content_view.read()
+    # Get published content-view version id
+    assert len(content_view.version) == 1
+    cvv = content_view.version[0].read()
+    assert len(cvv.environment) == 1
+    # Delete the content-view version from selected env
+    content_view.delete_from_environment(cvv.environment[0].id)
+    # Delete the version
+    content_view.version[0].delete()
+    # Make sure that content view version is really removed
+    assert len(content_view.read().version) == 0
+
+
+@pytest.mark.upgrade
+@pytest.mark.tier2
+def test_positive_delete_non_default():
+    """Create content view and publish and promote it to new
+    environment. After that try to disassociate content view from 'Library'
+    and one more non-default environments through 'delete_from_environment'
+    command and delete content view version from that content view.
+
+    :id: 95bb973c-ebec-4a72-a1b6-ad28b66bd11b
+
+    :expectedresults: Content view version deleted successfully
+
+    :CaseLevel: Integration
+
+    :CaseImportance: Critical
+    """
+    org = entities.Organization().create()
+    content_view = entities.ContentView(organization=org).create()
+    # Publish content view
+    content_view.publish()
+    content_view = content_view.read()
+    assert len(content_view.version) == 1
+    assert len(content_view.version[0].read().environment) == 1
+    lce = entities.LifecycleEnvironment(organization=org).create()
+    promote(content_view.version[0], lce.id)
+    cvv = content_view.version[0].read()
+    assert len(cvv.environment) == 2
+    # Delete the content-view version from selected environments
+    for env in reversed(cvv.environment):
+        content_view.delete_from_environment(env.id)
+    content_view.version[0].delete()
+    # Make sure that content view version is really removed
+    assert len(content_view.read().version) == 0
+
+
+@pytest.mark.upgrade
+@pytest.mark.tier2
+@pytest.mark.skipif((not settings.repos_hosting_url), reason='Missing repos_hosting_url')
+def test_positive_delete_composite_version():
+    """Create composite content view and publish it. After that try to
+    disassociate content view from 'Library' environment through
+    'delete_from_environment' command and delete content view version from
+    that content view. Add repository to initial content view
+    for better coverage.
+
+    :id: b5bb547e-0174-464c-b974-0254d372cdd6
+
+    :expectedresults: Content version deleted successfully
+
+    :CaseLevel: Integration
+
+    :BZ: 1276479
+    """
+    org = entities.Organization().create()
```

Comment:
```suggestion
```
Try not to create a new org as it takes time and places another load on Satellite under test
Random names should be enough to prevent tests in this file from interfering with each other